app/services/link_filter.rb

```diff
@@ -0,0 +1,29 @@
+class LinkFilter
+  attr_reader :result
+
+  def initialize(authenticated:, params:)
+    @authenticated = authenticated
+    @params = params
+  end
+
+  def execute
+    get_links
+  end
+
+private
+
+  def get_links
+    if @authenticated
+      search_result =
+        if @params[:tag].present?
+          Link.tagged_with(@params[:tag])
```

Comment:
i think we should also add a constraint for the links to check issue id
we should not show links without the issue